mobile_verifier/src/coverage.rs

```diff
@@ -0,0 +1,715 @@
+        })
+    }
+}
+
+#[derive(PartialEq, Debug)]
+pub struct CoverageReward {
+    pub cbsd_id: String,
+    pub points: Decimal,
+    pub hotspot: PublicKeyBinary,
+}
+
+#[allow(dead_code)]
+pub const MAX_RADIOS_PER_HEX: usize = 5;
+
+#[async_trait::async_trait]
+pub trait CoveredHexStream {
+    async fn covered_hex_stream<'a>(
+        &'a self,
+        cbsd_id: &'a str,
+        coverage_obj: &'a Uuid,
+        period_end: DateTime<Utc>,
+    ) -> Result<BoxStream<'a, Result<HexCoverage, sqlx::Error>>, sqlx::Error>;
+}
+
+#[derive(sqlx::FromRow)]
+pub struct Seniority {
+    pub uuid: Uuid,
+    pub seniority_ts: DateTime<Utc>,
+    pub last_heartbeat: DateTime<Utc>,
+    pub inserted_at: DateTime<Utc>,
+}
+
+#[async_trait::async_trait]
+impl CoveredHexStream for Pool<Postgres> {
+    async fn covered_hex_stream<'a>(
+        &'a self,
+        cbsd_id: &'a str,
+        coverage_obj: &'a Uuid,
+        period_end: DateTime<Utc>,
+    ) -> Result<BoxStream<'a, Result<HexCoverage, sqlx::Error>>, sqlx::Error> {
+        // Adjust the coverage
+        let seniority: Seniority = sqlx::query_as(
+            r#"
+            SELECT * FROM seniority
+            WHERE
+              cbsd_id = $1 AND
+              inserted_at <= $2
+            ORDER BY inserted_at DESC
+            LIMIT 1
+            "#,
+        )
+        .bind(cbsd_id)
+        .bind(period_end)
+        .fetch_one(self)
+        .await?;
+
+        // We can safely delete any seniority objects that appear before the latest in the reward period
+        sqlx::query("DELETE FROM seniority WHERE inserted_at < $1")
+            .bind(seniority.inserted_at)
+            .execute(self)
+            .await?;
+
+        // Find the time of insertion for the currently in use coverage object
+        let current_inserted_at: DateTime<Utc> = sqlx::query_scalar(
+            "SELECT inserted_at FROM hex_coverage WHERE cbsd_id = $1 AND uuid = $2 LIMIT 1",
+        )
+        .bind(cbsd_id)
+        .bind(coverage_obj)
+        .fetch_one(self)
+        .await?;
+
+        // Delete any hex coverages that were inserted before the one we are currently using, as they are
+        // no longer useful.
+        sqlx::query(
+            "DELETE FROM hex_coverage WHERE cbsd_id = $1 AND uuid != $2 AND inserted_at < $3",
+        )
+        .bind(cbsd_id)
+        .bind(coverage_obj)
+        .bind(current_inserted_at)
+        .execute(self)
+        .await?;
+
+        Ok(
+            sqlx::query_as("SELECT * FROM hex_coverage WHERE cbsd_id = $1 AND uuid = $2")
+                .bind(cbsd_id)
+                .bind(coverage_obj)
+                .fetch(self)
+                .map_ok(move |hc| HexCoverage {
+                    coverage_claim_time: seniority.seniority_ts,
+                    ..hc
+                })
+                .boxed(),
+        )
+    }
+}
+
+#[derive(Default)]
+#[allow(dead_code)]
```

Comment:
why all the dead code, is this in prep for phase 3?

Reply:
yes